src/server/helpers/file.py

```diff
@@ -45,3 +46,18 @@ def write_file(
     path = fs.get_path(fullpath)
 
     return path
+
+
+def create_file_filter(project: Project) -> Callable[[str], bool]:
+    fs = project.filesystem
+    fullpath = fs.get_fullpath(".gitignore")
+    if not fullpath.exists():
+        return lambda _: False
+
+    matches = parse_gitignore(fullpath)  # type: ignore
+
+    def ignore_path(path: str) -> bool:
+        fullpath = fs.get_fullpath(path)
+        return True if matches(fullpath) else False
```

Comment:
@shashigharti 
I don't think we need a fullpath here as gitignore has to check a relative path for match